src/Raven.Server/ServerWide/Context/TransactionOperationContext.cs

```diff
@@ -137,18 +152,32 @@ public void CloseTransaction()
 
         public override void Dispose()
         {
+            if (_locked.Raise() == false)
+            {
+                _continuation = Dispose;
```

Comment:
I know there is a race here. `Raise` check and `_continuation` set should be atomic, but it would make the code complex/harder to read for an extreme rare case.
In worst case we will leak a context.

Reply:
A user that use a compare exchange a lot may have a lot of those, they will eventually kill us, no?
Why aren't we using the `_continuation` itself as the lock? And having it as not null would do, no?